api.php

```diff
@@ -198,45 +209,80 @@
 
 //Change info for a user that already exists
 $app->post('/user/{id}', function (Request $request, Response $response, array $args) {
+
+	//set preg_match regex for name -will be used for first and last name
+	$nameRegEx = '/^[a-z ,.\'-]+$/i';
+
+	//regex pattern for phone number
+	$numberRegEx = '/\(?(\d{3})[\)\s-]*(\d{3})[\s\-]?(\d{4})/';
 	// get int value of requested id
 	$id = intval($args['id']);
 	// check validity of id
 	if (!$id) {
-		// set a message to explain what broke
-		$response->getBody()->write(json_encode([
-			'message' => 'invalid id provided',
-		]));
-		// return the error and a invalid request status
-		return $response
-			->withHeader('Content-Type', 'application/json')
-			->withStatus(400);
+		// set the response message
+		return badRequestResponse(['message' => 'invalid id provided'], $response);
 	}
 
 	// look up the user
 	$user = User::findUserById($id);
 	// if no matching user found
 	if (!$user) {
 		// set the response message
-		$response->getBody()->write(json_encode([
-			'message' => 'no user found for id:' . $id,
-		]));
-		// return the error and a 404 status
-		return $response
-			->withHeader('Content-Type', 'application/json')
-			->withStatus(404);
+		return notFoundResponse('no user found for id', $response);
 	}
 
 	// get the request as a stdObject
 	$body = json_decode($request->getBody());
 
 
-	//Update user info
+	//Update user info that already exists
 	if (isset($body->first_name)) {
-		$user->setFirstName($body->first_name);
+		$userFirstName = $body->first_name;
+		//check name input against string pattern 
+		if (!preg_match($nameRegEx, $userFirstName)) {
+			//set response for invalid name format
+			return badRequestResponse([
+				'field' => ['first_name'],
+				'message' => 'invalid character in first name'
+			], $response);
+		}
+
+		//length of first name
+		$stringLength = strlen($userFirstName);
+
+		//if length of firstName isn't greater than or equal to one or less than or equal to thirty return an error
+		if (!$stringLength >= 1 && !$stringLength <= 30) {
```

Comment:
I was able to update a name to be greater than 30 characters, but it did not accept unusual characters.

Reply:
should also be fixed as of last nights commits